### ISSUE TITLE
Question: Howto add custom command line options on v2.0.0

### ISSUE THREAD

**Issue by jardleex** [rundeck-plugins/ansible-plugin#67]
Hello,

I just updated to plugin version 2.0.0 while using Ansible 2.1.2.0 on Ubuntu 16.04.1.

In v1.4.0 there was an option to add custom command line options to an Playbook executed in a workflow step. This seems to be gone.
Is there a replacement or may it come back?

Thank you for your time and work

Jard

**Comment by yassineazzouz** [comment]
Hi @jardleex ,

Is there a specific option that you would like to add ? I believe most of the options provided in ansible are covered.
The whole idea behind adding all the option fields is to not have to add the options manually, and make some options like passwords and public key benefit from the functionalities offered by rundeck (like secure options and storage path ...) so adding the the old Extra Options fields will just make it more confusing, since the managed options can interfere with the ones defined manually.

Thanks,

**Comment by jardleex** [comment]
Hello @yassineazzouz,

thanks for your fast reply. 
I'd like to use `--tags` for example. Also playing around with `--limit` to find out how to couple with our existing roles and site.yml. A `-vvv` might be also nice sometimes.

Thank you.

**Comment by yassineazzouz** [comment]
So for --limit : there are a couple of ways you can do this:
1- In your Job target hosts : the nodes you select as your job target hosts, will be used to limit the execution.
2- In your project configuration : you can use project.ansible-limit in your project configuration file to limit   the jobs execution (at the project level)

Concerning -vvvv you can run the jobs on Debug output, or you can also use project.ansible-debug to have debug output enabled at the project level 

--tags is not included for now, but can easily be added :)

**Comment by jardleex** [comment]
`--limit` - **SOLVED**
Agreed. I played around with dispatch and was able to fulfill my needs. The other options at project or job level are good to know but dispatch is sufficient for now.

`-vvv`\- **WISHED**
I tried _project.ansible-debug_ but aint got the known Ansible verbose output in /var/log/rundeck/service.log. 
May it possible to add another option at a workflog/job level to optionally enable the debug output? Maybe a selector from 0-4 or similar.

`--tags`\- **WISHED**
Having this would be awesome. Our site.yml will grow in the next time and I'd like to be able to run only parts from it. On ~600 targets, this might save time.

**Comment by jardleex** [comment]
Thos two would be great to:
`--check` - **WISHED**
`--diff` - **WISHED**

When running playbooks via the Rundeck sheduler they would enable the user to see possible changes if the comitter ain't run the playbook after his change.

Some may also use `--list-(hosts|tags|tasks)` for debugging purposes as well.

As you see a custom command line options field wouldn't be that bad ;-).

**Comment by yassineazzouz** [comment]
Hi @jardleex ,

Concerning the other options I will check a way to have a custom ansible options field, I agree that it is easier to just have a "custom attributes" field than having every single cmd option in one separate option, although I still think that it could be a bit confusing to have some options managed automatically by the plugin and some others set in a separate field, plus typically rundeck is used for jobs that run in a production like environment, more for automating operational processes, not really for for debugging scripts. 

Concerning the debug output please use the job Log Level (screenshot below)
![debug](https://cloud.githubusercontent.com/assets/12769067/19322830/eb23ebcc-90ba-11e6-9a22-03851b3681ac.PNG)

Thanks,

**Comment by jardleex** [comment]
Hi @yassineazzouz,

I can understand the conflict I brought you in. Each dedicated option would need to check if it was also given in the custom options field. Or maybe Ansible takes care of this itself? E.g. having multiple `--tags` or `--check` may simply get merged by Ansible.

In version 1.4 I used dispatch and `--limit` at the same time to find out how the Plugin + Rundeck + Ansible work together. I'm pretty sure that my custom option has overwritten my dispatch selection (which is I think also only uses `--limit`).
You're right that the usage of Rundeck+Ansible is more of a daily use then for debugging purposes. Just wanted to mention it so in certain cases users may want to do debugging here. But thats not top prio.
Having the `--tags` would be much greater :)

Regarding the debug output:
You're right, now I got my verbose Ansible output in Rundeck. My assumption was that setting just **project.ansible-debug** would be sufficient was wrong. Thank's for the hint.

Best regards

Jard
